modules/api/php/endpoints/candidate/visit/instrument/flags.class.inc

```diff
@@ -127,6 +128,108 @@ class Flags extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PUT request.
+     * Creates or updates all statuses for a given instrument with new values.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request) : ResponseInterface
+    {
+        // TODO :: Check permissions. How??
```

Comment:
See comment on #7088

Reply:
are you sure you didn't mean your comment on #7075 ? I applied the change here, as it builds on this one.

Reply:
`data_entry `is already checked. The second point is resolved in the parent endpoint `modules/api/php/endpoints/candidate/visit/instrument/instrument.class.inc` (the part commented by Xavier below) , so a user without permission can't reach this function.